draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,76 +60,296 @@ collector). Thus it is up to each deployment of DAP to securely implement its
+task. In particular, aggregator should deserialize `vdaf_config` corresponding
+to `vdaf_type`, and pass the relevant parameters to the VDAF initializer. At
+this point, the task provision step has completed.
 
 # Leader Behavior
 
-> TODO(wangshan) Say what the Leader does when it receives a report with this
-> extension. What do Leaders do if they *don't* support this extension? Is it
-> safe to ignore it? How are the upload, aggregate, and collect flows impacted?
+Leader should have saved any parameters described in {{out-of-band-parameters}}.
+Leader may not know the task ID of the current task before receiving the first
+upload request.
+
+## Change to upload sub-protocol
+
+Upon receiving a report, leader reads the extension codepoint in
+`extension_type`. If the leader does not support this extension, it MUST ignore
+it. In particular, if the task ID is not known, then it MUST abort the handshake
+with "unrecognizedTask".
+
+> reason for aborting: if leader ignores it, then there is no way to ensure
+> clients that the extension is used and task parameters are respected in
+> aggregators
+
+If aggregator supports `task-prov` extension, it should proceed to
+{{provisioning-a-task}}. If task provision failed, leader MUST alert the client
+with error from {{provisioning-a-task}}. If task provision succeeded, leader
+should continue to the rest of upload flow.
+
+Leader MAY return error to the client if task creation failed.
 
+## Change to collect sub-protocol
+
+During collect init, leader must perform batch validation (see batch-validation
+in {{?DAP=I-D.draft-ietf-ppm-dap-01}}) to the `CollectReq`. This requires leader
+to know the parameters associated with the DAP task. However, upon receipt of
+`CollectReq`, leader may not know the parameters if the first upload has not
+arrived yet. In this case the leader MAY respond with HTTP status code 404
+Not Found and an error of type `unrecognizedTask`. The response MAY include a
+Retry-After header field to suggest a pulling interval to the collector.
+
+> OPEN ISSUE: Alternatively the leader can just response with 303 and continue
+> waiting.
 
 # Helper Behavior
 
-> TODO(wangshan) Say what the Helper does when it receives a report share with
-> this extension. (Same questions as above.)
+Helper should have saved any parameters described in {{out-of-band-parameters}}.
+Similar to leader, helper may not know the task ID of the current task before
+receiving the first `AggregateInitializeReq` message from the leader.
+
+## Change to helper aggregate sub-protocol
 
+Upon receipt of a `AggregateInitializeReq`, helper reads the extension
+codepoint in `extension_type`. If helper does not support this extension, it
+MUST abort the aggregate protocol and alert the leader with error
+"unrecognizedMessage".
+
+> OPEN ISSUE: Alternatively Helpr MUST ignore the unrecognized extension, see
+> [#334](https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/334) for
+> discussion.
```

Comment:
Fine to point to #334, but I think for this PR we should align the Helper's behavior with the Leader's. In particular we should say that it MUST ignore the extension.